syntax = "proto3";
package iroha.protocol;
import "commands.proto";
import "primitive.proto";

message Header {
  uint64 created_time = 1;
  repeated Signature signatures = 2;
}

message Transaction {

  message Meta {
    string creator_account_id = 1;
    // used to prevent replay attacks.
    uint64 tx_counter = 2;
  }

  message Body {
    repeated Command commands = 1;
  }

  Header header = 1;
  Meta meta = 2;
  Body body = 3;
}

message Block {


  message Meta {
    uint32 tx_number = 1; // the number of transactions inside. Maximum 16384 or 2^14
    uint64 height = 2; // the current block number in a ledger
    bytes merkle_root = 3; // global merkle root
    bytes prev_block_hash = 4; // Previous block hash
  }

  message Body {
    repeated Transaction transactions = 1;
  }


  Header header = 1;
  Meta meta = 2;
  Body body = 3;
}